test/server.js

```diff
@@ -52,29 +55,25 @@ var fileReadInterval = setInterval(() => {
   });
 }, 1000);
 
-
-// prometheus should scrape this endpoint for data.
-server.get('/metrics', (req, res) => {
-  console.log(`GET /metrics ${register.metrics()}`);
-  res.set('Content-Type', register.contentType);
-  res.end(register.metrics());
-});
-
-server.get('/protractor', (req, res) => {
-  var sys = require('util');
-  var exec = require('child_process').exec;
+var smokeTestsInterval = setInterval(() => {
+  console.log("[INFO] Running protractor tests in interval");
   var child;
-  console.log("[INFO] Running protractor tests");
   child = exec("protractor /protractor/protractor.conf.js", function (error, stdout, stderr) {
-    // sys.print('stderr: ' + stderr);
     if (error !== null) {
       console.log('[ERROR] ' + error);
-    } else {
-      res.send("[INFO] Test triggered.");
+      // Not sure if we want to stop testing after an error occurs
+      // clearInterval(smokeTestsInterval);
     }
     console.log('[INFO] ' + stdout);
   });
-  res.end();
+}, env.test_interval);
+
+
+// prometheus should scrape this endpoint for data.
+server.get('/metrics', (req, res) => {
```

Comment:
Wondering now, per @spadgett's comment, if the `/metrics` enpoint also needs a rate limit to ensure no DOS attack.

Reply:
@benjaminapetersen definitely would be god to have the limit there as well.